python/src/numerous/collection/numerous_document.py

```diff
@@ -15,44 +15,57 @@ class NumerousDocument:
     Attributes
     ----------
         key (str): The key of the document.
-        collection_id (str): The id of collection document belongs to.
-        data (Optional[Dict[str, Any]]): The data of the document.
+        collection_info tuple[str, str]: The id
+        and key of collection document belongs to.
+        data (Optional[dict[str, Any]]): The data of the document.
         id (Optional[str]): The unique identifier of the document.
-        cleint (Client): The client to connect.
-        tags (Dict[str, str]): The tags associated with the document.
+        client (Client): The client to connect.
+        tags (dict[str, str]): The tags associated with the document.
 
     """
 
     def __init__(
         self,
         client: Client,
         key: str,
-        collection_id: str,
+        collection_info: tuple[str, str],
         numerous_doc_ref: Optional[CollectionDocumentReference] = None,
     ) -> None:
         self.key: str = key
-        self.collection_id: str = collection_id
+        self.collection_id: str = collection_info[0]
+        self.collection_key: str = collection_info[1]
         self._client: Client = client
         self.document_id: Optional[str] = None
 
         if numerous_doc_ref is not None:
             dict_of_tags = {tag.key: tag.value for tag in numerous_doc_ref.tags}
-            self.data: Optional[Dict[str, Any]] = base64_to_dict(numerous_doc_ref.data)
+            self.data: Optional[dict[str, Any]] = base64_to_dict(numerous_doc_ref.data)
```

Comment:
I think this should be defined above as 
```python
self.data: Optional[dict[str, Any]] = None
```
to avoid potential attribute errors, right?